src/backend.rs

```diff
@@ -82,7 +82,14 @@ pub trait Connector: Send + Sync {
     /// "Self::is_valid" would fail, the connections in the pool will
     /// churn between "connecting" and "failing health checks".
     ///
-    /// This function must be cancel-safe.
+    /// ## Cancel Safety
+    ///
+    /// The future returned by `connect` may be cancelled by the `qorb` pool if
+    /// the slot is dropped. If this occurs and a connection has already been
+    /// established, the connection should be closed. If the connection is not
+    /// yet established, the attempt to establish it should also be cancelled.
+    /// In both cases, any cleanup logic required by the type of connection
+    /// should be performed when the `connect` future is dropped.
```

Comment:
this is great, thank you! my one additional question is, is "slot" terminology exposed to the user of `qorb` elsewhere? I don't remember OTTOMH whether that's an internal implementation detail or something the user is expected to know what we mean when we say it.